pygdf/_gdf.py

```diff
@@ -92,6 +92,23 @@ def apply_mask_and(col, mask, out):
     return len(out) - nnz
 
 
+def gdf_to_np_dtype(dtype):
+    """Util to convert gdf dtype to numpy dtype.
+    """
+    return np.dtype({
+         libgdf.GDF_FLOAT64: np.float64,
+         libgdf.GDF_FLOAT32: np.float32,
+         libgdf.GDF_INT64: np.int64,
+         libgdf.GDF_INT32: np.int32,
+         libgdf.GDF_INT16: np.int16,
+         libgdf.GDF_INT8: np.int8,
+         libgdf.GDF_INT8: np.bool_,
+         libgdf.GDF_DATE64: np.datetime64,
+         libgdf.N_GDF_TYPES: np.int32,
+         libgdf.GDF_CATEGORY: np.int32,
```

Comment:
categoricals are currently `int8` and in another PR we are dealing with them by `if pd.core.common.is_categorical_dtype(dtype):
        return libgdf.GDF_INT8`. This mapping to `np.int32` is problematic

Reply:
@dantegd I think this may be the start of the change for categoricals where we use a hash instead of an enumeration. Would significant changes need to be made to the category class to support that?